proto/src/chrono.rs

```diff
@@ -1,53 +0,0 @@
-use core::convert::TryInto;
-
-use chrono::{DateTime, Duration, TimeZone, Utc};
-
-use crate::google::protobuf as pb;
-
-impl From<DateTime<Utc>> for pb::Timestamp {
-    fn from(dt: DateTime<Utc>) -> pb::Timestamp {
-        pb::Timestamp {
-            seconds: dt.timestamp(),
-            // This can exceed 1_000_000_000 in the case of a leap second, but
-            // even with a leap second it should be under 2_147_483_647.
-            nanos: dt
-                .timestamp_subsec_nanos()
-                .try_into()
-                .expect("timestamp_subsec_nanos bigger than i32::MAX"),
-        }
-    }
-}
-
-impl From<pb::Timestamp> for DateTime<Utc> {
-    fn from(ts: pb::Timestamp) -> DateTime<Utc> {
-        Utc.timestamp(ts.seconds, ts.nanos as u32)
```

Comment:
This panicked if members of `ts` have invalid values (subject to `chrono`'s interpretation which does not readily conform to the protobuf specification).

As the most likely origin of `pb::Timestamp` is by having been parsed from an incoming gRPC message by prost without any validation as to its contents, this is an easy DoS bomb under any application developers who would make use of this conversion. Please consider using fallible conversions when dealing with untrusted values originating from the network.